ethpm/utils/backend.py

```diff
@@ -65,9 +72,14 @@ def get_resolvable_backends_for_uri(
     default_ipfs = get_ipfs_backend_class()
     if default_ipfs in URI_BACKENDS and default_ipfs().can_resolve_uri(uri):
         yield default_ipfs
-    for backend_class in URI_BACKENDS:
-        if backend_class is default_ipfs:
-            continue
-        # type ignored because of conflict with instantiating BaseURIBackend
-        elif backend_class().can_resolve_uri(uri):  # type: ignore
-            yield backend_class
+    else:
+        for backend_class in URI_BACKENDS:
+            if backend_class is default_ipfs:
+                continue
+            # type ignored because of conflict with instantiating BaseURIBackend
+            else:
+                try:
+                    if backend_class().can_resolve_uri(uri):  # type: ignore
+                        yield backend_class
+                except ConnectionError:
+                    logger.info("No local IPFS node available on port 5001.")
```

Comment:
@pipermerriam I'm not super familiar with the logging module, is this more or less the way to go about it?

Reply:
```python
import logging

logger = logging.getLogger('ethpm.utils.backend')


def my_thing(...):
    logger.info("This is a message that will be logged at info level")
    logger.debug("This one is at debug")
    # https://github.com/ethereum/snake-charmers-tactical-manual/issues/19
    logger.info("We use percent formatting with logging messages: %s", some_arg)
```

That's all you have to do.

Reply:
For this specific case you can do:

```python
try:
    ...
except Something:
    logger.debug("Something went wrong", exc_info=true)
```

This will include a full stacktrace in the logging message which can be nice when debugging these things.  Not entirely sure that's the right approach here but probably doesn't hurt.